File: Frontend/bookmyplay/src/services/bookingService.js
import api from "./api";

export const createBooking = (booking) => {
    return api.post("/bookings", booking);
};

export const getMyBookings = (userId) => {
    return api.get(`/bookings/user/${userId}`);
};

export const cancelBooking = (bookingId) => {
    return api.put(`/bookings/${bookingId}/cancel`);
};

export const getBookingDetails = (id) => {
    return api.get(`/bookings/${id}`);
};

export const rescheduleBooking = (id, newSlotId) => {
    return api.put(`/bookings/reschedule/${id}?newSlotId=${newSlotId}`);
};

File: Backend/bookmyplay-api/src/main/java/com/bookmyplay/service/BookingService.java
package com.bookmyplay.service;

import com.bookmyplay.dto.BookingResponse;
import com.bookmyplay.dto.CreateBookingRequest;
import java.util.List;

public interface BookingService {

    BookingResponse createBooking(CreateBookingRequest request);

    List<BookingResponse> getBookingsByUser(Long userId);

    String cancelBooking(Long bookingId);

    BookingResponse getBookingById(Long bookingId);

    String rescheduleBooking(Long bookingId, Long newSlotId);

    List<BookingResponse> getBookingsByVendor(Long vendorId);
}